engine/resources/net_test.go

```diff
@@ -0,0 +1,168 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"golang.org/x/sys/unix"
+)
+
+// test cases for NetRes.unitFileContents()
+var unitFileContentsTests = []struct {
+	in  *NetRes
+	out []byte
+}{
+	{
+		&NetRes{
+			BaseRes: BaseRes{
+				Name: "eth0",
```

Comment:
@purpleidea How do I set the name, now that BaseRes is gone?

Reply:
`engine.NewNamedResource`

https://godoc.org/github.com/purpleidea/mgmt/engine#NewNamedResource

Then if you want to set properties on it you need to cast it to a *NetRes and then set fields as you wish. We should probably add a helper function to do this with `reflect`. Patches welcome.